test/load/loadtests.py

```diff
@@ -43,40 +82,94 @@ class LoadTest(TestCase):
+        which_test()
+
+    def test_login_session_flow(self):
+        """Do a full login-flow with cert signing etc."""
+        # Login as either a new or existing user.
+        if self._perc(PERCENT_LOGIN_CREATE):
             session = self._authenticate_as_new_user()
+            can_delete = True
         else:
             session = self._authenticate_as_existing_user()
-        # Fetch keys and make some number of signing requests.
-        session.fetch_keys()
-        session.check_session_status()
-        session.get_random_bytes()
-        num_sign_reqs = random.randint(SIGN_REQS_MIN, SIGN_REQS_MAX)
-        for i in xrange(num_sign_reqs):
-            session.sign_certificate(DUMMY_PUBLIC_KEY)
-        # Teardown the session, and maybe the whole account.
-        session.destroy_session()
-        if random.randint(0, 100) < ACCOUNT_DELETE_PERCENT:
-            self.client.destroy_account(
-                email=session.email,
-                stretchpwd=self._get_stretchpwd(session.email),
-            )
+            can_delete = False
+        try:
+            # Do a whole lot of account status polling.
+            n_poll_reqs = random.randint(LOGIN_POLL_REQS_MIN,
+                                         LOGIN_POLL_REQS_MAX)
+            for i in xrange(n_poll_reqs):
+                session.get_email_status()
+            # Always fetch the keys.
+            session.fetch_keys()
+            # Sometimes check the session status.
+            if self._perc(PERCENT_LOGIN_STATUS):
+                session.check_session_status()
+            # Sometimes get some random bytes.
+            if self._perc(PERCENT_LOGIN_RANDBYTES):
+                session.get_random_bytes()
+            # Always do some number of signing requests.
+            n_sign_reqs = random.randint(LOGIN_SIGN_REQS_MIN,
+                                         LOGIN_SIGN_REQS_MAX)
+            for i in xrange(n_sign_reqs):
+                session.sign_certificate(DUMMY_PUBLIC_KEY)
+            # Sometimes tear down the session.
+            if self._perc(PERCENT_LOGIN_TEARDOWN):
+                session.destroy_session()
+        except fxa.errors.ClientError as e:
+            # There's a small chance this could fail due to concurrent
+            # password change destroying the session token.
+            if e.errno != ERROR_INVALID_TOKEN:
+                raise
+        # Sometimes destroy the account.
+        if can_delete:
+            if self._perc(PERCENT_LOGIN_CREATE_DESTROY):
+                self.client.destroy_account(
+                    email=session.email,
+                    stretchpwd=self._get_stretchpwd(session.email),
+                )
 
     def _get_stretchpwd(self, email):
         return hashlib.sha256(email).hexdigest()
 
+    def _get_new_user_email(self):
+        uid = uniq()
+        return "loads-fxa-{}-new@restmail.lcip.org".format(uid)
+
+    def _get_existing_user_email(self):
+        uid = random.randint(1, 1)#999)
```

Comment:
I'm assuming the '#999)' is a for-debug-only hack that snuck into the final patch, right?